### PR TITLE
Only run update logic if objects are actually different

### PR DESCRIPTION
**Pull request by HlinaCZ**
Reflects https://github.com/openshift/cluster-network-operator/pull/160

### CODE REVIEW

**Comment by ovirt-infra** [comment]
Hello contributor, thanks for submitting a PR for this project! 

I am the bot who triggers "standard-CI" builds for this project.
As a security measure, I will not run automated tests on PRs that are not from white-listed contributors.

In order to allow automated tests to run, please ask one of the project maintainers to review the code and then do one of the following:

1. Type `ci test please` on this PR to trigger automated tests for it.
2. Type `ci add to whitelist` on this PR to trigger automated tests for it and also add you to the contributor white-list so that your future PRs will be tested automatically. ( keep in mind this list might be overwritten if the job XML is refreshed, for permanent whitelisting, please follow #3 option )
3. If you are planning to contribute to more than one project, maybe it's better to ask them to add you to the project organization, so you'll be able to run tests for all the organization's projects.

**Comment by SchSeba** [comment]
Hi @HlinaCZ Thanks for the contribution can you please run dep ensure looks like you add a new dependence in the project

File: pkg/apply/apply.go [hunk 1]
```diff
@@ -51,14 +52,12 @@ func ApplyObject(ctx context.Context, client k8sclient.Client, obj *uns.Unstruct
...
 	}
-
-	if err := client.Update(ctx, obj); err != nil {
-		return errors.Wrapf(err, "could not update object %s", objDesc)
-	}
-	if existing.GetResourceVersion() == obj.GetResourceVersion() {
-		log.Print("update was noop")
-	} else {
-		log.Print("update was successful")
+	if !equality.Semantic.DeepEqual(existing, obj) {
```
Comment (SchSeba) [inline pkg/apply/apply.go]: I am not sure this function achive the objective.
can you please add a unit  test for this.

Comment (phoracek) [reply pkg/apply/apply.go]: Uh, that will be hard. We need mocking for this function, but can try

Comment (SchSeba) [reply pkg/apply/apply.go]: It's not must I can live with that

**Comment by phoracek** [comment]
/hold

Unit tests WIP

**Comment by phoracek** [comment]
ci add to whitelist

**Comment by phoracek** [comment]
ci test please

File: pkg/apply/apply.go [hunk 2]
```diff
@@ -51,14 +52,12 @@ func ApplyObject(ctx context.Context, client k8sclient.Client, obj *uns.Unstruct
...
+		if err := client.Update(ctx, obj); err != nil {
+			return errors.Wrapf(err, "could not update object %s", objDesc)
+		} else {
+			log.Print("update was successful")
```
Comment (SchSeba) [inline pkg/apply/apply.go]: should this be a debug log?

Comment (phoracek) [reply pkg/apply/apply.go]: The logger we currently use does not handle levels.

**Comment by phoracek** [comment]
@SchSeba merging is not a part of this PR. It is checked in merge_test.go. In this PR we care whether the object has not been touched by update (version remains the same) or it has been touched (minimal check is looking at annotations, the actual merging process is tested in the merge_test.go)

**Comment by phoracek** [comment]
Thanks @qinqon

**Comment by phoracek** [comment]
Thanks @HlinaCZ

**Pull request merged**